arangod/Aql/grammar.y

```diff
@@ -1390,7 +1390,8 @@ upsert_statement:
             std::string_view nodeMemberName = nodeMember->getStringView();
             if (nodeMemberName == arangodb::StaticStrings::IndexHintOption || 
                 nodeMemberName == arangodb::StaticStrings::IndexHintOptionForce ||
-                nodeMemberName == arangodb::StaticStrings::IndexHintDisableIndex) {
+                nodeMemberName == arangodb::StaticStrings::IndexHintDisableIndex ||
+                nodeMemberName == arangodb::StaticStrings::UseCache) {
```

Comment:
What is the reasoning behind this flag?
A user can switch of the cache usage on demand? Is there a use-case for it?

Reply:
Yes, the use cache is: I have a hash cache for an index, but for some reason I know it is not going to help.
I think it is always good to provide an option so users can opt out if necessary. Nobody is forced to use the option, but if users complain about bad performance for some queries, and we find the cache to cause problems, we have a way to turn it off easily on a per-query basis.